File: CPP/virtual/p11.java

class A  {



	 	 void getName1(int x) {
			System.out.println( "hello A");

		}





};

class B extends A{


	 	 void getName1(float x) {
			System.out.println( "hello B");

		}


};

class Test{

	public static void main(String args[]){
			B ptr = new B();
			
			ptr.getName1(10.3f);
			ptr.getName1(11);

		
	}

}

File: CPP/virtual/p11.cpp
#include<iostream>
using namespace std;

//#overloading does not take place in inheritence in cpp
class A  {

	public:

	 	 const char* getName1(int x) {
			return "hello A";

		}





};

class B:public A{

	public:



		 const char* getName1(float x) {
			 cout<<x<<endl;
			return "hello B";

		}

};

int main(){


	B b;


	cout<<b.getName1(10.5f)<<endl;
	cout<<b.getName1(1)<<endl;



}


